backend/neofs/neofs.go

```diff
@@ -0,0 +1,804 @@
+		default:
+			if value != "" {
+				headers[key] = value
+			}
+		}
+	}
+
+	if headers[object.AttributeTimestamp] == "" {
+		headers[object.AttributeTimestamp] = strconv.FormatInt(src.ModTime(ctx).UTC().Unix(), 10)
+	}
+
+	rawObj := formRawObject(f.pool.OwnerID(), cnrID, filepath.Base(containerPath), headers)
+	p := new(client.PutObjectParams).WithObject(rawObj.Object()).WithPayloadReader(in)
+
+	oid, err := f.pool.PutObject(ctx, p)
+	if err != nil {
+		return nil, err
+	}
+
+	p2 := new(client.ObjectHeaderParams).WithAddress(newAddress(cnrID, oid))
+	obj, err := f.pool.GetObjectHeader(ctx, p2)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, id := range ids {
+		dp := new(client.DeleteObjectParams).WithAddress(newAddress(cnrID, id))
+		_ = f.pool.DeleteObject(ctx, dp)
+	}
+
+	return newObject(f, obj, ""), nil
+}
+
+func (o *Object) Update(ctx context.Context, in io.Reader, src fs.ObjectInfo, options ...fs.OpenOption) error {
+	obj, err := o.fs.Put(ctx, in, src, options...)
+	if err != nil {
+		return err
+	}
+
+	objInfo, ok := obj.(*Object)
+	if !ok {
+		return fmt.Errorf("invalid object type")
+	}
+
+	if err = o.Remove(ctx); err != nil {
+		fs.Logf(o, "couldn't remove old file after update '%s': %s", o.OID(), err.Error())
+	}
+
+	o.filePath = objInfo.filePath
+	o.remote = objInfo.remote
+	o.name = objInfo.name
+	o.timestamp = objInfo.timestamp
+	o.Object = objInfo.Object
+
+	return nil
+}
+
+func (o *Object) Remove(ctx context.Context) error {
+	p := new(client.DeleteObjectParams).WithAddress(newAddress(o.ContainerID(), o.OID()))
+	return o.fs.pool.DeleteObject(ctx, p)
+}
+
+func (f *Fs) NewObject(ctx context.Context, remote string) (fs.Object, error) {
+	containerStr, containerPath := bucket.Split(filepath.Join(f.root, remote))
+
+	cnrID, err := f.parseContainer(containerStr)
+	if err != nil {
+		return nil, fs.ErrorDirNotFound
+	}
+
+	ids, err := f.findObjectsFilePath(ctx, cnrID, containerPath)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(ids) == 0 {
+		return nil, fs.ErrorObjectNotFound
+	}
+
+	param := new(client.ObjectHeaderParams).WithAddress(newAddress(cnrID, ids[0]))
+	obj, err := f.pool.GetObjectHeader(ctx, param)
+	if err != nil {
+		return nil, err
+	}
+
+	return newObject(f, obj, ""), nil
+}
+
+func (f *Fs) Mkdir(ctx context.Context, dir string) error {
+	containerStr, _ := bucket.Split(path.Join(f.root, dir))
+	if containerStr == "" {
+		return nil
+	}
+
+	cnrID, err := f.parseContainer(containerStr)
+	if err == nil {
+		return nil
+	}
+
+	pp, err := policy.Parse(f.opt.PlacementPolicy)
+	if err != nil {
+		return err
+	}
+
+	cnr := container.New(
+		container.WithPolicy(pp),
+		container.WithOwnerID(f.pool.OwnerID()),
+		container.WithCustomBasicACL(f.opt.BasicACL),
+		container.WithAttribute(container.AttributeName, containerStr),
+		container.WithAttribute(container.AttributeTimestamp, strconv.FormatInt(time.Now().UTC().Unix(), 10)))
+
+	container.SetNativeName(cnr, containerStr)
```

Comment:
Container per directory is a viable option, but it may be not quite suitable for backups:
- container creation is a paid operation, 
- backups may need custom eACL on a predefined container,
- restic backend uses different approach.

Maybe we can treat directories based on the filepath of actual files? To create `/new/dir`, restic may create "hidden" file, e.g. `/new/dir/.neofs.hidden` which will be manually ignored in the listing?

/cc @realloc

Reply:
Actually we create only one (root) container. Inside it when we do list we already treat directories based on the filepath. And I don't see any profit of creating fiction dir files in such case.

Reply:
Okay, that's great, then. I thought `Mkdir` might be called for every subdir in object file paths.